### DISCUSSION TITLE
Stop / Remove TinyPilot

### DISCUSSION THREAD

**Discussion by petterpet** [tiny-pilot/tinypilot#451] [Q&A] [answered]
Is there a possibility to stop tinypilot?
If I stop tinypilot service (`sudo service tinypilot stop`) there is just something in the background using the video-input device (the video is shown in the browser althought the connection indicator is red). How can I disable this?

And is it possible to uninstall tinypilot with all its services?

**Comment by mtlynch** [comment, marked as answer]
Unfortunately, there is no supported way to remove everything automatically. The video service is called ustreamer, so you can disable that with `sudo service ustreamer stop`.

To completely reverse all of TinyPilot's installation, one would need to review [ansible-role-tinypilot](https://github.com/mtlynch/ansible-role-tinypilot) and manually undo the configuration changes it applies.

**Comment by mataebi** [comment]
Here is a collection of commands that have worked here to clean a machine so that it was possible reinstall TinyPilot's standard installation script without unexpected problems while (re)installing everything (no failed tasks). You might need to adjust it here and there to fit your environment, but over all it should work on most platforms.
```
service tinypilot stop
service ustreamer stop
service janus stop
service usb-gadget stop

systemctl disable tinypilot.service
systemctl disable ustreamer.service
systemctl disable janus.service
systemctl disable usb-gadget.service

rm /usr/lib/systemd/system/tinypilot.*
rm /usr/lib/systemd/system/ustreamer.*
rm /usr/lib/systemd/system/janus.* 
rm /usr/lib/systemd/system/usb-gadget.*

reboot

apt purge -y tinypilot
apt purge -y janus
apt autoremove -y

deluser --remove-home --remove-all-files tinypilot
deluser --remove-home --remove-all-files ustreamer

rm -rf /opt/tinypilot* /opt/ustreamer*
rm -rf /sys/class/udc/musb-hdrc.?.auto
rm -rf /sys/kernel/config/usb_gadget
rm /boot/overlay-user/h3-cc-otg*

rm -rf /etc/janus
rm -rf /usr/share/janus
rm -rf /usr/lib/aarch64-linux-gnu/janus/plugins
rm /var/cache/apt/archives/janus-dev_1.0.1-1~bpo11+1_arm64.deb
rm /var/cache/apt/archives/janus_1.0.1-1~bpo11+1_arm64.deb

rm /etc/nginx/sites-enabled/tinypilot.conf
```
